week-6/Homework/mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,7 +1,40 @@
 // Implement the missing functionality according to the task description.
 // Hint: Remember which function you need to call when it's time to alarm.
+// Global variables
+let setbtn = document.getElementById("set");
+let stopbtn = document.getElementById("stop");
+let display = document.getElementById("timeRemaining");
+let timeValue = document.getElementById("alarmSet");
+let timer;
+
+// This function will have a counter in seconds and will change the display by deducting a second everytime it's called. when the counter is 0 it will call playAlarm function
+function trigger() {
+  let counter = parseInt(display.innerText.slice(0,2) * 60) + parseInt(display.innerText.slice(3));
```

Comment:
Thought about this when I revisited the code To correct for a negative value, will try to implement it this week

Reply:
It is also good to handle invalid numbers like a negative number.

What I meant is not to read the value from the `display.innerText` but read it from a variable instead. Only read the the `display` element once when you click on the `Set Alarm` button to initiate the alarm clock. 

E.g.
```
let counter = parseInt(timeValue.value);
// then when trigger is called, you would only
//  - deduct one from the counter (counter--)
//  - and set display (display.innerText = ....)
trigger();
```

Reply:
When I tried to define the counter outside of the trigger funtion it had no value assigned to it as it grabs the value as soon as the page is loaded I suppose.

Reply:
it would look stg like this:
 - `let counter = 0;`
 - then in your `setAlarm` function you have to update it `counter = parseInt(timeValue.value);`
 - and finally your `trigger` function will be decreasing it.